csharp/src/Apache.Arrow/Column.cs

```diff
@@ -62,7 +62,7 @@ private bool ValidateArrayDataTypes()
         {
             for (int i = 0; i < Data.ArrayCount; i++)
             {
-                if (Data.Array(i).Data.DataType != Field.DataType)
+                if (Data.Array(i).Data.DataType.TypeId != Field.DataType.TypeId)
```

Comment:
Don't we need also compare other members?

* `Children` when it is `NestedType`
* `ByteWidth` when it is `FixedSizeBinaryType`

I suppose we need to create a new `ArrowType` comparer for the product projects like `ArrayTypeComparer` of `Apache.Arrow.Tests` and use it.
https://github.com/apache/arrow/blob/master/csharp/test/Apache.Arrow.Tests/ArrayTypeComparer.cs

Reply:
Sure, I'll try to work on it.

Reply:
Hi @HashidaTKS,

Please review below updates:

Added \src\Apache.Arrow\Arrays\ArrayDataTypeComparer.cs 
Added unit testing to cover most of the code paths

Thanks,

Reply:
Ah, my comment was not good.

* `ArrayDataTypeComparer` need to support `TimestampType`, `Date32Type`, `Date64Type`,  `Time32Type`, `Time64Type`, `FixedSizeBinaryType`. `ListType` and `StructType` like `Apache.Arrow.Tests.ArrayTypeComparer.cs`.
* How about using Visitor Pattern for `ArrayDataTypeComparer`?
  * https://github.com/apache/arrow/blob/master/csharp/README.md#implementation
  * > Uses Acyclic Visitor Pattern for array types and arrays to facilitate serialization, record batch traversal, and format growth.

Reply:
Sure, I have used Visitor Pattern for ArrayDataTypeComparer, please review.